crates/sui-storage/src/lock_service.rs

```diff
@@ -76,6 +92,12 @@ struct LockServiceImpl {
     /// the lock once it is set. After a certificate for this object is processed it can be
     /// forgotten.
     transaction_lock: DBMap<ObjectRef, Option<TransactionDigest>>,
+
+    /// The semantics of transaction_lock ensure that certificates are always processed
+    /// in causal order - that is, certificates naturally form a partial order. tx_sequence
+    /// records a total ordering among all processed certificates (which is naturally local
+    /// to this authority).
+    tx_sequence: DBMap<TransactionDigest, TxSequenceNumber>,
```

Comment:
it should be safe, yes - the main purpose of this map is to ensure we don't sequence a tx twice. Because transactions can't cross epochs, a tx digest can't appear in this map in two different epochs with two different sequences.